ui/src/hooks/useHalo2BinaryMerkleTree.ts

```diff
@@ -0,0 +1,33 @@
+import { useWorker } from '@/hooks/useWorker'
+import { type GenerateMerkleProofFn, Halo2BinaryMerkleTreeWorker } from '@anonklub/halo2-binary-merkle-tree-worker'
+
+export const useHalo2BinaryMerkleTreeWorker = () => {
+  const isWorkerReady = useWorker(Halo2BinaryMerkleTreeWorker)
+
+  const generateMerkleProof: GenerateMerkleProofFn = async (
+    leaves,
+    leaf,
+    depth,
+  ): Promise<Uint8Array> => {
+    process.env.NODE_ENV === 'development' && console.time('==>merkle')
+
+    // eslint-disable-next-line no-useless-catch
+    try {
```

Comment:
Now I am confused.
Here we need to keep using a useless try catch but in other places not?

Reply:
Yes you need to be confused, I just had in my mind to remove this try/catch today. :sweat_smile: 
I tested it already now and it works as expected.